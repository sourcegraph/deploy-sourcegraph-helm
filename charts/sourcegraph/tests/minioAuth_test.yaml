---
suite: minioAuth
templates:
- minio/minio.Secret.yaml
- minio/minio.Deployment.yaml
tests:
- it: should reference existing secret name when existingSecret is passed
  template: minio/minio.Deployment.yaml
  set:
    minio:
      auth:
        existingSecret: "my-secret"
  asserts:
      - equal:
          path: spec.template.spec.containers[0].env[0].name
          value: MINIO_ROOT_USER
      - equal:
          path: spec.template.spec.containers[0].env[0].valueFrom.secretKeyRef.name
          value: "my-secret"
      - equal:
          path: spec.template.spec.containers[0].env[1].name
          value: MINIO_ROOT_PASSWORD
      - equal:
          path: spec.template.spec.containers[0].env[1].valueFrom.secretKeyRef.name
          value: "my-secret"
      - matchSnapshot:
            path: spec.template.metadata.annotations.[checksum/minio.secret]
- it: should not generate a secret when existingSecret is passed
  template: minio/minio.Secret.yaml
  set:
    minio:
      auth:
        existingSecret: "my-secret"
  asserts:
      - hasDocuments:
          count: 0
- it: should generate a secret when existingSecret is blank
  template: minio/minio.Secret.yaml
  set:
    minio:
      auth:
        existingSecret: ""
  asserts:
      - hasDocuments:
          count: 1
- it: should generate a secret by default
  template: minio/minio.Secret.yaml
  asserts:
      - hasDocuments:
          count: 1
- it: should use provided value in secret
  template: minio/minio.Secret.yaml
  set:
    minio:
      auth:
        password: "hello"
  asserts:
      - equal:
          path: data.password
          value: "aGVsbG8="
- it: should reference default secret name when existingSecret is blank
  template: minio/minio.Deployment.yaml
  asserts:
      - equal:
          path: spec.template.spec.containers[0].env[0].name
          value: MINIO_ROOT_USER
      - equal:
          path: spec.template.spec.containers[0].env[0].valueFrom.secretKeyRef.name
          value: "minio-auth"
      - equal:
          path: spec.template.spec.containers[0].env[1].name
          value: MINIO_ROOT_PASSWORD
      - equal:
          path: spec.template.spec.containers[0].env[1].valueFrom.secretKeyRef.name
          value: "minio-auth"
      - matchSnapshot:
          path: spec.template.metadata.annotations.[checksum/minio.secret]
