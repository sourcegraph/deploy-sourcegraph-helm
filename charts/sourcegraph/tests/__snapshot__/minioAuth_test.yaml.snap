should reference default secret name when existingSecret is blank:
  1: |
    50eb78907271d137c54240e04157c250d923ba7c9332c7af51a21b86c9bcf486
should reference existing secret name when existingSecret is passed:
  1: |
    f6de451a224b164adfd2a9d5f6f7d3c7be5ff312b0397de2a1ed6aa25544cf88
